runsc/sandbox/network.go

```diff
@@ -326,6 +332,112 @@ func createInterfacesAndRoutesFromNS(conn *urpc.Client, nsPath string, conf *con
 	return nil
 }
 
+func prepareInitStackArgs(args *boot.InitExternalStackArgs, nsPath string) error {
+	// Join the network namespace that we will be copying.
+	restore, err := joinNetNS(nsPath)
+	if err != nil {
+		return err
+	}
+	defer restore()
+	// Get all interfaces in the namespace.
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return fmt.Errorf("error querying interfaces: %v", err)
+	}
+	if isRoot, _ := isRootNS(); isRoot {
+		return fmt.Errorf("cannot run in with network enabled in root network namespace")
+	}
+	// Collect addresses and routes from the interfaces.
+	for _, iface := range ifaces {
+		if iface.Flags&net.FlagUp == 0 {
+			log.Infof("Skipping down interface: %+v", iface)
+			continue
+		}
+		// Skip loopback devices.
+		if iface.Flags&net.FlagLoopback != 0 {
+			continue
+		}
+		allAddrs, err := iface.Addrs()
+		if err != nil {
+			return fmt.Errorf("error fetching interface addresses for %q: %v", iface.Name, err)
+		}
+		// Keep only IPv4 addresses.
```

Comment:
Why keep only IPv4 addresses?

Reply:
Sorry for confusion. It is specific for TLDK only since we only support IPv4 in TLDK for now. To make it more general, we should remove this check here and support both IPv4 and IPv6. Thanks for pointing out.